project2-beat-that/script.js

```diff
@@ -1,4 +1,266 @@
+// [3.1] how score will be tabulated: if input is 1, dice 1 and dice 2 will be concatenated; if input is 2, dice 2 and dice 1 will be concatenated.
+// [4] Repeat steps 1 to 3 for player 2.
+// [5] Program will finally compare the scores and show result.
+// [5.1] list out all possible scenarios (win, lose, draw).
+
+var gameMode = "player1diceroll";
+
+//initialize array to store 2 dices rolled.
+var dicesRolled = [];
+// var message = " ";
+
+//initialize first player.
+var currentPlayer = 1;
+
+// Player 1 and Player 2 score.
+var player1Score = 0;
+var player2Score = 0;
+
+//randomly generate a number  from 1 to 6
+var generateRandomNumber = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//push the dice number of each of 2 rolls into the array
+var get2DiceRolls = function () {
+  dicesRolled = [];
+  for (var i = 0; i < 2; i += 1) {
+    dicesRolled.push(generateRandomNumber());
+  }
+  return dicesRolled;
+};
+
+var getPlayer1Score = function (userInput) {
+  let firstDice = dicesRolled[0];
+  let secondDice = dicesRolled[1];
+  if (userInput === 1) {
+    player1Score = Number(String(firstDice) + String(secondDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+  if (userInput === 2) {
+    player1Score = Number(String(secondDice) + String(firstDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player1Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
+
+var getPlayer2Score = function (userInput) {
+  let firstDice = dicesRolled[0];
+  let secondDice = dicesRolled[1];
+  if (userInput === 1) {
+    player2Score = Number(String(firstDice) + String(secondDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player2Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+  if (userInput === 2) {
+    player2Score = Number(String(secondDice) + String(firstDice));
+    return `Based on the dice you've rolled, the sum of your score is ${player2Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  input = Number(input);
+  // var gameMode;
+  if (gameMode === "player1diceroll") {
+    dicesRolled = get2DiceRolls();
+    gameMode = "player1diceorder";
+    return `Welcome to the game, player ${currentPlayer}. <br><br>
+    You rolled 2 dices. <br>
+    Dice 1: ${dicesRolled[0]}. <br><br>
+    Dice 2: ${dicesRolled[1]}. <br><br>
+    Now, ENTER either 1 or 2 BELOW and click the blue buzzer ABOVE to define the order of your dice roll. This program will then show you your score.
+    `;
+  }
+
+  if (gameMode === "player1diceorder") {
+    if (input != 1 && input != 2) {
+      return `Please only ENTER the number 1 or 2 BELOW. Then, click on the buzzer to check your score.`;
+    }
```

Comment:
good input validation, but in this case, i might have forgotten what my dice rolls are

would help from a user experience pov to show the dice rolls to me again so i know what score i will get if i were to enter 1 or 2